### ISSUE TITLE
Log request (and response) in single batch

### ISSUE THREAD

**Issue by apandichi** [OpenFeign/feign#1035]
Feign's Logger is logging once per request / response item.
For example, JavaLogger will log the following request / response pair by invoking java.util.logging.Logger 7 times:

```
[SendsStuff#login] ---> POST http://localhost:61113/ HTTP/1.1
[SendsStuff#login] Content-Length: 80
[SendsStuff#login] Content-Type: application/json
[SendsStuff#login] ---> END HTTP (80-byte body)
[SendsStuff#login] <--- HTTP/1.1 200 OK (1ms)
[SendsStuff#login] content-length: 3
[SendsStuff#login] <--- END HTTP (3-byte body)
```

In a Kibana + Logstash environment, these logs are currently scattered by having one document per log line. Which means the request logs are scattered in different documents. The same goes for the response logs. It's making difficult to follow the logs.

What I'm suggesting is to have some sort of log batching such that we end up with one log call for the request, another for the response.  Having a single log per request/response means having a single document for request, another for response. That would be much easier to follow.

**Comment by kdavisk6** [comment]
@apandichi 

Thank you for submitting this request.  We try to include enhancements were this significant community support.  I've marked this issue as a proposal and if it receives enough support, we will accept this as an enhancement.  

This is similar to #941.  We may want to combine these proposals into a consolidated logging api update.  Would you be interesting in putting that together?

**Comment by dharmeshjogadia** [comment]
@kdavisk6 I totally agree with @apandichi . We have the same requirement on the addition of the Log level controller. Currently, `Slf4jLogger` logging request and response as `DEBUG`. Can we have a config to set the Log level for same?

**Comment by finnetrolle** [comment]
Guys, do we really want that strange format where one big text is splitted by several log lines? I think it is harder to parse. May be it's better to have such log?
```
[main] INFO TestAPI - 
---> GET http://view-localhost:52859/default/api/foo HTTP/1.1
---> END HTTP (0-byte body)
<--- HTTP/1.1 200 OK (33ms)content-length: 3
foo
<--- END HTTP (3-byte body)
```

**Comment by velo** [comment]
Well, the information we use to assemble this log comes out from multiple locations.... to do as you suggested, we would need to collect logs "somewhere", aggregate, and print on any termination event (could be response being properly processed, could be on http error, could be on NPE on a decoder written by a 3rd party).

I admit it would look cool, but man, that would be hard to do.
